core/inject.js

```diff
@@ -232,19 +232,19 @@ Blockly.createMainWorkspace_ = function(svg, options, blockDragSurface,
   mainWorkspace.translate(0, 0);
   Blockly.mainWorkspace = mainWorkspace;
 
-  if (!options.readOnly && !options.hasScrollbars) {
+  if (!options.readOnly && !mainWorkspace.isContentBounded_()) {
     var workspaceChanged = function(e) {
-      if (!mainWorkspace.isDragging()) {
+      // We always check isContentBounded_ again because the original
+      // "not bounded" state of isContentBounded_ could have been changed.
+      if (!mainWorkspace.isDragging() && !mainWorkspace.isContentBounded_()) {
         var metrics = mainWorkspace.getMetrics();
-        var edgeLeft = metrics.viewLeft + metrics.absoluteLeft;
-        var edgeTop = metrics.viewTop + metrics.absoluteTop;
-        if (metrics.contentTop < edgeTop ||
+        if (metrics.contentTop < metrics.viewTop ||
             metrics.contentTop + metrics.contentHeight >
-            metrics.viewHeight + edgeTop ||
+            metrics.viewHeight + metrics.viewTop ||
             metrics.contentLeft <
-                (options.RTL ? metrics.viewLeft : edgeLeft) ||
+                (options.RTL ? metrics.viewLeft : metrics.viewLeft) ||
```

Comment:
why don't remove this condition?

Reply:
Yep, I need to take another look at this whole function. Coordinates man :/

Thanks for pointing it out to me!

Reply:
Turns out that I thought the "+ metrics.absoluteLeft" and "+ metrics.absoluteTop" were causing a problem, but the problem was actually that the scrollX and scrollY were not getting set after setMetrics was called, so viewTop and viewLeft were getting calculated incorrectly. This has been fixed.

Everything (except the mainworkspace.isContentBounded_() checks) has been reverted, and the workspaceChanged function is working once again.